sympy/printing/tests/test_pycode.py

```diff
@@ -70,6 +79,29 @@ def test_pycode_reserved_words():
     assert py_str in ('else_ + if_', 'if_ + else_')
 
 
+def test_sqrt():
+    prntr = PythonCodePrinter()
+    assert prntr._print_Pow(sqrt(x), rational=False) == 'math.sqrt(x)'
+    assert prntr._print_Pow(sqrt(x), rational=True) == 'x**(1/2)'
+
+    prntr = MpmathPrinter()
+    assert prntr._print_Pow(sqrt(x), rational=False) == 'mpmath.sqrt(x)'
+    assert prntr._print_Pow(sqrt(x), rational=True) == \
+        'x**(mpmath.mpf(1)/mpmath.mpf(2))'
+
+    prntr = NumPyPrinter()
+    assert prntr._print_Pow(sqrt(x), rational=False) == 'numpy.sqrt(x)'
+    assert prntr._print_Pow(sqrt(x), rational=True) == 'x**(1/2)'
```

Comment:
I'm not sure about these others -- if they are running in Python, how will you be sure that you aren't getting truncation as a result of 1/2 -> 0?

Reply:
I think that it had been printing rationals without considering py2 compatibility for a long time.